wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/devfile/DevfileVersion.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.workspace.server.devfile;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import javax.inject.Singleton;
+import org.eclipse.che.api.workspace.server.devfile.exception.DevfileException;
+
+/** Class that helps determine devfile versions. */
+@Singleton
+public class DevfileVersion {
```

Comment:
Such name is hardly giving idea that this is an utility class. Sounds more like a POJO or data object for me.
Can we add an verb, like DevfileVersionDetector\Extractor\Retriever\... ?

Reply:
`DevfileVersionDetector` now